### PR TITLE
Use rspec seed to make SAMPLE consistent

### PR DESCRIPTION
**Pull request by stefkin**
### What is the purpose of this pull request?

It makes sample of test cases consistent with seed.

It should help to reproduce issues which may appear during SAMPLE run.

Also helps in a sutation where you want to measure sample group iteratively. 

### What changes did you make? (overview)

I create a random number generator using rspec's seed and use it in Array#sample call

I couldn't figure out how to test this for minitest (it doesn't have example specific output), suggestions are welcome! 

### Is there anything you'd like reviewers to focus on?

Please check out my tests. Are 10 runs ok to make sure that seed works consistently?

### Checklist

- [x] I've added tests for this change
- [x] I've added a Changelog entry
- [ ] I've updated a documentation

### CODE REVIEW

**Comment by stefkin** [comment]
I figured out how to test minitest in a similar way (using -v option).

But in order to get minitest's seed I'd need to hook into plugin_test_prof.

Would it be ok to move TestProf::MinitestSample#run logic into the plugin?

**Comment by palkan** [comment]
> But in order to get minitest's seed I'd need to hook into plugin_test_prof. Would it be ok to move TestProf::MinitestSample#run logic into the plugin?

Can we leave the sampling logic where it is and add a `TestProf::MinitestSample.random_generator` instead which we can populate from the plugin? Will it work?

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## master (unreleased)
 
+- SAMPLE and SAMPLE_GROUP work consistently with seed in RSpec. ([@stefkin][])
```
Comment (palkan) [inline CHANGELOG.md:5]: Please, add a shortcut with the URL to the bottom of the file. Currently, it's not clickable: https://github.com/palkan/test-prof/blob/903649336e49ec042aa85d6a8530499396ae14a0/CHANGELOG.md

Comment (stefkin) [reply CHANGELOG.md:5]: done, thanks

**Pull request merged**